hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieStorageConfig.java

```diff
@@ -130,6 +130,12 @@ public class HoodieStorageConfig extends HoodieConfig {
       .defaultValue("TIMESTAMP_MICROS")
       .withDocumentation("Sets spark.sql.parquet.outputTimestampType. Parquet timestamp type to use when Spark writes data to Parquet files.");
 
+  public static final ConfigProperty<String> PARQUET_FIELD_ID_WRITE_ENABLED = ConfigProperty
+      .key("hoodie.parquet.fieldId.write.enabled")
+      .defaultValue("true")
+      .withDocumentation("Sets spark.sql.parquet.fieldId.write.enabled. "
+          + "If enabled, Spark will write out parquet native field ids that are stored inside StructField's metadata as parquet.field.id to parquet files.");
```

Comment:
.sinceVersion("0.12.0")?

Reply:
Good point. This field is added in Spark 3.3 . It doesn't break older Spark but it would not be effective. I'll add this into `withDocumentation`